src/Altinn.App.Core/Models/Layout/PageComponentConverter.cs

```diff
@@ -446,6 +446,34 @@ private BaseComponent ReadComponent(ref Utf8JsonReader reader, JsonSerializerOpt
         return new BaseComponent(id, type, dataModelBindings, hidden, required, readOnly, additionalProperties);
     }
 
+    private Dictionary<string, ModelBinding>? DeserializeModelBindings(ref Utf8JsonReader reader)
+    {
+        var modelBindings = new Dictionary<string, ModelBinding>();
+        if (reader.TokenType != JsonTokenType.StartObject)
+        {
+            throw new JsonException("Expected StartObject token for \"dataModelBindings\"");
+        }
+
+        while (reader.Read() && reader.TokenType != JsonTokenType.EndObject)
+        {
+            if (reader.TokenType != JsonTokenType.PropertyName)
+            {
+                throw new JsonException();
+            }
+
+            var propertyName = reader.GetString()!;
+            reader.Read();
+            modelBindings[propertyName] = reader.TokenType switch
+            {
+                JsonTokenType.String => new ModelBinding { Field = reader.GetString()! },
```

Comment:
IMO we shouldn't use null-forgiving operator this much, because it's unclear why it should/could be null (and generally detracts from the point of NRT). Some options I could think of

* Make sure we use nullability annotations and null-checks (and related attributes (NotNullWhen etc)) correctly
* throw an exception with a better message as opposed to getting nullref
* use `Debug.Assert` as a last resort if we can't convince the compiler something is nonnull (but should always be nonnull)

Reply:
Totally agree, but in this case it’s the Utf8JsonReader that have an api that can’t be properly annotated, so there is not much we can do.